### DISCUSSION TITLE
HASP Initializing (no more)

### DISCUSSION THREAD

**Discussion by JAAlperin** [HASwitchPlate/HASPone#191] [Troubleshooting]
Two year old device purchased complete.   Updated to HASPone 1.05.  Can't get to work.  Hangs at "HASP Initializing"

Ran core blueprint but got no automations for plate02.  Can access via url:
MQTT Status: Connected
MQTT ClientID: plate02-382b784489
HASPone FW Version: 1.05
LCD Model:
LCD FW Version: 0
LCD Active Page: 6
LCD Serial Speed: 921600
CPU Frequency: 160MHz
Sketch Size: 661728 bytes
Free Sketch Space: 2482176 bytes
Heap Free: 18392
Heap Fragmentation: 9
ESP core version: 3.0.2
IP Address: 192.168.0.156
Signal Strength: -56
Uptime: 1418
Last reset: Power On

MQTT received following.  Note LCD not connected.
Message 747 received on hasp/plate02/sensor at 6:42 AM:
{
    "espVersion": 1.05,
    "updateEspAvailable": false,
    "lcdConnected": false,
    "lcdVersion": "0",
    "updateLcdAvailable": false,
    "espUptime": 60,
    "signalStrength": -54,
    "haspName": "plate02",
    "haspIP": "192.168.0.156",
    "haspClientID": "plate02-382b784489",
    "haspMac": "38:2b:78:4:48:9",
    "haspManufacturer": "HASwitchPlate",
    "haspModel": "HASPone v1.0.0",
    "heapFree": 21912,
    "heapFragmentation": 21,
    "heapMaxFreeBlockSize": 16784,
    "espCore": "3.0.2"
}

I had many false starts with blueprints due to yaml configuration.  Finally got coexistence with these entries:
## only used for the GUI editors
automation: !include automations.yaml
## All of my regular yaml automations go here
automation yaml: !include_dir_merge_list automations/

But, blueprint doesn't seem to fully run (I do get the lovelace notice).

Any suggestions would be appreciated.

**Comment by aderusha** [comment]
Did the device work prior to the update?  If so, try setting the serial speed to 115200 and see if there's any difference in behavior.

**Comment by JAAlperin** [comment]
Thanks aderusha.  Yes, device was operational 18 months ago.
Shortly after posting I updated Nexion from chip, reset, and lowered speed.  Eventually raised it again.  That got the device somewhatt working.  Edits to blueprint are not retained. After running the actions the device will lose wifi connection and reports MQTT failed.  Remove power, restore, OK.  But page changes take 10 - 30 seconds.  I'll experiment further.

**Comment by JAAlperin** [comment]
Not usable.  Connects, disconnects, MQTT fails, brings up HASwitchPlate setup, reboots.  Sometimes http available, sometimes not. Won't calm down.

1653494640: New connection from 192.168.0.156:56252 on port 1883.
1653494651: Client <unknown> closed its connection.
1653494651: New connection from 192.168.0.156:58251 on port 1883.
1653494651: New client connected from 192.168.0.156:58251 as plate02-382b784489 (p2, c1, k30, u'MQTTuser').
1653494744: Client plate02-382b784489 has exceeded timeout, disconnecting.
1653494753: New connection from 192.168.0.156:55380 on port 1883.
1653494846: Client <unknown> has exceeded timeout, disconnecting.
1653494877: New connection from 192.168.0.156:53180 on port 1883.
1653494879: New client connected from 192.168.0.156:53180 as plate02-382b784489 (p2, c1, k30, u'MQTTuser').
1653494885: New connection from 192.168.0.156:59731 on port 1883.
1653494885: Client plate02-382b784489 already connected, closing old connection.
1653494885: New client connected from 192.168.0.156:59731 as plate02-382b784489 (p2, c1, k30, u'MQTTuser').
1653494891: New connection from 192.168.0.156:49385 on port 1883.
1653494891: Client plate02-382b784489 already connected, closing old connection.
1653494891: New client connected from 192.168.0.156:49385 as plate02-382b784489 (p2, c1, k30, u'MQTTuser').
1653494897: New connection from 192.168.0.156:65436 on port 1883.
1653494897: Client plate02-382b784489 already connected, closing old connection.
1653494897: New client connected from 192.168.0.156:65436 as plate02-382b784489 (p2, c1, k30, u'MQTTuser').
1653494902: Saving in-memory database to /data//mosquitto.db.
1653494926: New connection from 192.168.0.156:59009 on port 1883.
1653494937: New connection from 192.168.0.156:49889 on port 1883.
1653494939: Client plate02-382b784489 already connected, closing old connection.
1653494939: New client connected from 192.168.0.156:49889 as plate02-382b784489 (p2, c1, k30, u'MQTTuser').

**Reply by ielbury** [reply]
How many WiFi devices do you have on your network and what Router/AP are you using? 

The reason why I ask is I had a LOT of WiFi related issues until I upgraded my Router//AP to a WiFi 6 model. Many of the older consumer WiFi Router/AP's can only support 20-30 devices max. In my case I had ~ 60 devices across a Router + repeater and I had the same problems you were having. 

Once I upgraded to a TP-Link AX73 model which supports ~ 200 devices, my WiFi network has been rock solid since the upgrade and I did not require a repeater anymore. The bottom line is you may be at the edge of your routers 2.4G WiFi capacity.

Is your WiFi network split and using separate 2.4G and 5G networks? I recently had a problem with a defective ESP device that was taking down the 2.4G network but everything on the 5G network worked fine. I never thought that one defective WiFi node could take down the whole 2.4G network! That was like looking for a needle in a haystack. My only grace was I was able to identify the date/time the problems started and then was able to use ESPHome RSSI values to identify the device that failedbecause it's RSSI value became unavailable.

After I had that issue with the defective ESP, one of my plates would not work properly, it started by not being able to connect, then I was able to connect but is displayed a blank web page when using the admin site. Reflashing the plate did nothing and I had to do a http://192.168.0.XX/resetConfig to get it to work again.

**Reply by JAAlperin** [reply]
My "Router" is Raspberry Pi 4B 4GB Running OpenWRT.  Can handle thousands of devices simultaneously.  I only run a few hundred, maybe 100 active at any moment.  I use a few different access points to the Pi.  None are Gen 6.  Multiple 2G and 5G nets.  I have zero problems except the HASPone.  Old HASP version running fine.

**Reply by ielbury** [reply]
That's awesome ... just sharing my problems and resolutions. I was running OpenWRT and my issues were with the Atheros/Broadcom chipsets not the OS. 

You seem technical, I would try the /resetConfig trick.

**Comment by aderusha** [comment]
Hey @JAAlperin - I've been working on the next release and wonder if maybe the current dev build might help you out.  Go to the "update firmware" page and enter the following URL for "Update ESP from URL": `https://raw.githubusercontent.com/HASwitchPlate/HASPone/dev-1.06/Arduino_Sketch/HASwitchPlate.ino.d1_mini.bin`

There have been some changes to the WiFi connectivity which might address what you're seeing above.  Let me know how it goes!

**Reply by JAAlperin** [reply]
Thanks, but actually worse.  When the device rebooted after the firmware update, the icon for the clock was missing.  Then it went into a connecting / disconnected loop the cycle for which is about 3 seconds.  Of course, the web UI is no longer available.

**Reply by JAAlperin** [reply]
Bricked.  Can't flash it offline.  "For parts" now.

**Reply by JAAlperin** [reply]
Does anybody have accurate directions for flashing the Wemos offline?  I have previously done this, but am now unable.

Alternatively, anybody want to buy my parts?

**Reply by ielbury** [reply]
You clearly don't like what I have to say but I will try help you one last time.

Download ESPHome Flasher from https://github.com/esphome/esphome-flasher/releases/tag/1.4.0 . This is a wrapper around NodeMCU Pyflasher in the official instructions that has all the defaults preset for you.

![image](https://user-images.githubusercontent.com/240860/171062225-c7eec973-78b2-4842-ac78-8ffdb6c1d3ba.png)

1. Refresh the ports.
2. Select the correct port. If you don't see the port then you need the appropriate serial port driver. Either CP2102 https://www.silabs.com/developers/usb-to-uart-bridge-vcp-drivers or CH34X https://github.com/nodemcu/nodemcu-devkit/tree/master/Drivers . My D1 mini's have the CH34X USB-Serial converters so I would start there.
3. Select your bin file and 
4. Click Flash ESP and you will see the flashing sequence in the Console window.

After the flash is complete, the Console will display the startup and initialization log from the serial port.

I HIGHLY recommend that when you get WiFi going, either the AP or connected to your router, enter the URL http://192.168.4.1/resetConfig and reset your plates configuration.

One other thing is to press the reset button on the D1 mini after pressing the "View Logs" button on ESP Home Flasher and watch the boot sequence in the Console. You can determine a lot about what is going on with the firmware to diagnose your problem.

**Reply by ielbury** [reply]
If you open the plate, circled in Red is the D1 Mini and 1) is the USB mini port that can be used to flash + power the device and 2) is the reset button.

![image](https://user-images.githubusercontent.com/240860/171067028-9e89f154-4f32-4751-81ed-f9a986da0a94.png)

You will need to pull the board out of the case to plugin the USB mini cable and have the CH34X drivers installed. 

The D1 does not require external power as USB will power it.

Your board may look like this

![image](https://user-images.githubusercontent.com/240860/171067489-5a665f32-e5ed-4de9-a0e5-e5a03c8810f7.png)

**Reply by aderusha** [reply]
I have a guide in the wiki for how to flash the device, but man the pictures above are great!  Here's the guide: https://github.com/HASwitchPlate/HASPone/wiki/Flashing-HASPone-to-an-ESP8266

**Comment by ielbury** [comment]
@aderusha I would add the /resetConfig to your FAQ. I have had to use it twice now to reset a plate that had a corrupt config file. The problem for me was, I could not even see the "factory reset settings" button in both cases.

**Comment by JAAlperin** [comment]
Thank you Ian.  I do like what you say, but it is not working at this time.

I have used Pyflasher, esphome, and Tasmotizer.  None are currently working even though they have in the past.  They all time out (no response from device).

I should have asked for wiring instructions.  I’m using a YP-05 that has successfully flashed a number of nodemcu devices previously.  I’m connecting tx>rx, rx>tx, gnd to gnd and 3v to 3v.  Com5 shows up.  The YP-05 is actively attempting connection without success.  Are any other connections needed (I’ve tried pulling D3/Flash both up and down with no change in results).  I’ve read a lot about other pins as well.

You mention pressing a reset button.  My device has two holes andthe word reset beside the upper hole, but no button.  My device is at least two years old, possibly older.

Thanks again for trying to help.  I would appreciate anything else you might suggest.

Regards,

Jeff Alperin
H: 561-666-7789
C: 561-797-9876

Ask Oscar Leroy about growing older

**Reply by ielbury** [reply]
The D1 mini is programmed using it's onboard CH340 USB to Serial gateway. See the picture's I added to my original instructions above.

**Comment by JAAlperin** [comment]
Thanks, Allen.  My Wemos has neither a USB port nor a reset button.  This is an old unit bought assembled.

I have been connecting a YP-05 tx>rx, rx>tx, gnd>gnd, and 3v>3v, with D3 disconnected, pulled low, or pulled high.  Wemos refuses to connect to esphome flasher or tasmotizer flasher.

What happens if I apply 3v to the Wemos and touch a screw driver to the reset holes?

You say “pull the board out of the case”, but my board is has two fasteners.  What size/type are they? (Looks like Torx).

Regards,

Jeff Alperin
H: 561-666-7789
C: 561-797-9876

Ask Oscar Leroy about growing older

**Reply by aderusha** [reply]
Ahhh OK I see your problem - the USB port and reset switch are on the bottom of the board!  You can't see them as you haven't removed the PCB yet, which is required to run the flash process.  So, what you'll need is a 1.5mm hex head "allen" wrench, take those two screws out, pull the PCB assembly out of the rear housing, and then you'll see the USB port that you're after.

**Reply by aderusha** [reply]
Hey @JAAlperin I wanted to check in and see how it's going for you - any improvement? Anything I can help with?

**Comment by JAAlperin** [comment]
Hi Allen.  Thanks for asking.  Nothing you can do at the moment.

Was able to completely reflash ESP to 1.05 and Nextion and configure device.  However, after I run the initial setup automation the device becomes flakey (takes 20+ seconds for a page change).  As previously reported, 1.06 enters a reboot loop.

Regards,

Jeff Alperin
H: 561-666-7789
C: 561-797-9876

Ask Oscar Leroy about growing older

**Reply by aderusha** [reply]
If you run a continuous ping against your HASP IP address, are you seeing any packet loss or high latency (> 10-ish ms)?

**Comment by JAAlperin** [comment]
Yes.

Takes 10+ seconds to change a page.
Ping results:
> quote_type
> PING 192.168.0.156 (192.168.0.156): 56 data bytes
> 64 bytes from 192.168.0.156: icmp_seq=0 ttl=255 time=12.867 ms
> 64 bytes from 192.168.0.156: icmp_seq=1 ttl=255 time=4.908 ms
> Request timeout for icmp_seq 2
> Request timeout for icmp_seq 3
> Request timeout for icmp_seq 4
> 64 bytes from 192.168.0.156: icmp_seq=5 ttl=255 time=34.998 ms
> Request timeout for icmp_seq 6
> Request timeout for icmp_seq 7
> Request timeout for icmp_seq 8
> 64 bytes from 192.168.0.156: icmp_seq=9 ttl=255 time=230.062 ms
> Request timeout for icmp_seq 10
> Request timeout for icmp_seq 11
> Request timeout for icmp_seq 12
> 64 bytes from 192.168.0.156: icmp_seq=13 ttl=255 time=50.482 ms
> Request timeout for icmp_seq 14
> 64 bytes from 192.168.0.156: icmp_seq=15 ttl=255 time=431.105 ms
> Request timeout for icmp_seq 16
> Request timeout for icmp_seq 17
> Request timeout for icmp_seq 18
> Request timeout for icmp_seq 19


Regards,

Jeff Alperin
H: 561-666-7789
C: 561-797-9876

Ask Oscar Leroy about growing older

**Comment by JAAlperin** [comment]
Meanwhile, on the same HASS, my plate01 (version 0.37 with a lot of mods) runs perfectly and pings continuously in about 4 ms.

Regards,

Jeff Alperin
H: 561-666-7789
C: 561-797-9876

Ask Oscar Leroy about growing older

**Comment by JAAlperin** [comment]
I factory reset the device and switched to HASwitchPlate network.  The I pinged 192.168.4.1.  No lost packets.  Consistent 4 ms response.  Then configured it for wifi on a different network.  Bad results again.  So I updated firmware successfully (network url). Ping results again bad.

HARDWARE?

Regards,

Jeff Alperin
H: 561-666-7789
C: 561-797-9876

Ask Oscar Leroy about growing older

**Comment by JAAlperin** [comment]
Yes.  I assumed that, but couldn’t find an Allen wrench to fit.  Finally found a screwdriver blade that fit.
Flashed 1.06.  Got the boot loop.
Flashed 1.05.  Page changes can take 30 seconds.  Text not as edited into automation (from blueprint).  Edits are not saved.
Oh well. I’ll take a look again in the future.

Meanwhile, my plate01 running 0.37, works perfectly.

Thanks for your help with the flashing.

Regards,

Jeff Alperin
H: 561-666-7789
C: 561-797-9876

Ask Oscar Leroy about growing older
